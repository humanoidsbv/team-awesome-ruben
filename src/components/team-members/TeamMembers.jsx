import React from 'react';

import TeamMemberAdd from '../team-member-add/';

const TeamMembers = () => {
  return (
    <React.Fragment>
      <TeamMemberAdd />
    </React.Fragment>
  );
};

export default TeamMembers;
