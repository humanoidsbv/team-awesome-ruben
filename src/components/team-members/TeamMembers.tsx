import React, { useEffect, useState } from 'react';

import IconArrowDown from '../../assets/icons/icon-arrow-down.svg';
import IconPlus from '../../assets/icons/icon-plus.svg';
import TeamMember from '../team-member/';
import TeamMemberAdd from '../team-member-add';
import styles from './TeamMembers.module.css';

import { TeamMemberInterface } from '../../redux/team-members/types';

export interface TeamMembersProps {
  addTeamMember: ({}) => void;
  fetchTeamMembers: () => void;
  orderToggle: boolean;
  sortTeamMembersByField: (value: string) => void;
  sortTeamMembersOrder: () => void;
  teamMembers: TeamMemberInterface[];
}

const TeamMembers = ({
  addTeamMember,
  fetchTeamMembers,
  orderToggle,
  sortTeamMembersByField,
  sortTeamMembersOrder,
  teamMembers
}: TeamMembersProps): React.ReactElement => {
  const [isFormVisible, setIsFormVisible] = useState(false);

  useEffect(() => {
    fetchTeamMembers();
  }, []);

  const handleFormVisible = (): void => setIsFormVisible(!isFormVisible);

  const handleChange = (event: React.ChangeEvent<HTMLSelectElement>): void =>
    sortTeamMembersByField(event.currentTarget.value);

  const handeClick = () => sortTeamMembersOrder();

  return (
    <React.Fragment>
      {isFormVisible && (
        <TeamMemberAdd
          addFormData={addTeamMember}
          handleFormVisible={handleFormVisible}
        />
      )}

      <div className={styles.header}>
        <span className={styles.heading}>All Humanoids</span>
        <button
          className={`${styles.addMember} ${isFormVisible &&
            styles.addMemberVisible}`}
          onClick={handleFormVisible}
          disabled={isFormVisible === true}
          type="button"
        >
          <IconPlus className={styles.addNewIcon} />
          New Humanoid
        </button>
        <select className={styles.sortMembers} onChange={handleChange}>
          <option value="firstName">First name</option>
          <option value="lastName">Last name</option>
          <option value="locality">Location</option>
          <option value="role">Project role</option>
          <option value="currentClient">Client name</option>
          <option value="startingDate">Date</option>
        </select>
        <button
          className={`${styles.sortMembersOrder} ${orderToggle &&
            styles.sortMembersActive}`}
          onClick={handeClick}
          type="button"
        >
          <IconArrowDown />
        </button>
      </div>
      {teamMembers.map(teamMember => (
        <TeamMember teamMember={teamMember} key={teamMember.id} />
      ))}
    </React.Fragment>
  );
};

export default TeamMembers;
