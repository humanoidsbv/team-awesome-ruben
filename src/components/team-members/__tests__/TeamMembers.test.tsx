import * as React from 'react';
import { shallow } from 'enzyme';

import TeamMembers from '../TeamMembers';

const mockedTeamMembersData = {
  currentClient: '',
  employeeNumber: '',
  firstName: '',
  id: 0,
  lastName: '',
  role: '',
  startingDate: ''
};

it('matches output with  snapshot', () => {
  const teamMembers = shallow(
    <TeamMembers
      addTeamMember={() => {}}
      fetchTeamMembers={() => {}}
      orderToggle={false}
      sortTeamMembersByField={() => {}}
      sortTeamMembersOrder={() => {}}
      teamMembers={[mockedTeamMembersData]}
    />
  );
  expect(teamMembers).toMatchSnapshot();
});

// addFormData={() => {}} handleFormVisible={() => {}}
