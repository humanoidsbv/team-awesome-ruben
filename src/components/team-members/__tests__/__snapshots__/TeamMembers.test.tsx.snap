// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches component output with expected output 1`] = `
<Fragment>
  <div
    className="header"
  >
    <span
      className="heading"
    >
      All Humanoids
    </span>
    <button
      className="addMember false"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <test-file-stub
        className="addNewIcon"
      />
      New Humanoid
    </button>
    <select
      className="sortMembers"
      onChange={[Function]}
    >
      <option
        value="firstName"
      >
        First name
      </option>
      <option
        value="lastName"
      >
        Last name
      </option>
      <option
        value="locality"
      >
        Location
      </option>
      <option
        value="role"
      >
        Project role
      </option>
      <option
        value="currentClient"
      >
        Client name
      </option>
      <option
        value="startingDate"
      >
        Date
      </option>
    </select>
    <button
      className="sortMembersOrder false"
      onClick={[Function]}
      type="button"
    >
      <test-file-stub />
    </button>
  </div>
  <TeamMember
    key="0"
    teamMember={
      Object {
        "currentClient": "",
        "employeeNumber": "",
        "firstName": "",
        "id": 0,
        "lastName": "",
        "role": "",
        "startingDate": "",
      }
    }
  />
</Fragment>
`;
